common/test/acceptance/tests/studio/test_studio_outline.py

```diff
@@ -492,6 +621,167 @@ def test_unchanged_release_date_is_not_saved(self):
+        """
+        Restrict unit access to a certain group and confirm that a
+        warning is displayed.  Then, remove the access restriction
+        and verify that the warning no longer appears.
+        """
+        self.assertFalse(unit.has_restricted_warning)
+        unit.set_unit_access('Content Groups', [self.content_group_a_id])
+        self.assertTrue(unit.has_restricted_warning)
+        unit.set_unit_access('Content Groups', [self.content_group_a_id])
+        self.assertFalse(unit.has_restricted_warning)
+
+    def test_units_can_be_restricted(self):
+        """
+        Visit the course outline page, restrict access to a unit.
+        Verify that there is a restricted group warning.
+        Remove the group access restriction and verify that there
+        is no longer a warning.
+        """
+        self.course_outline_page.visit()
+        self.course_outline_page.expand_all_subsections()
+        unit = self.course_outline_page.section_at(0).subsection_at(0).unit_at(0)
+        self._set_restriction_on_unrestricted_unit(unit)
+
+    def test_restricted_sections_do_not_appear_to_blocked_users_in_lms(self):
+        """
+        Test that a section restricted to one content group does not
+        appear to users outside of that content group.
+        """
+        self.course_outline_page.visit()
+        self.course_outline_page.expand_all_subsections()
+        unit = self.course_outline_page.section_at(0).subsection_at(0).unit_at(0)
+        unit.set_unit_access('Content Groups', [self.content_group_a_id])
+        self.course_outline_page.view_live()
+
+        course_home_page = CourseHomePage(self.browser, self.course_id)
+        course_home_page.visit()
+        course_home_page.resume_course_from_header()
+        self.assertEqual(course_home_page.outline.num_units, 2)
+        course_home_page.visit()
+        course_home_page.preview.set_staff_view_mode("Learner in Test Group B")
+        course_home_page.resume_course_from_header()
+        self.assertEqual(course_home_page.outline.num_units, 1)
+
+    def test_restricted_sections_do_appear_to_restricted_users_in_lms(self):
+        """
+        Verify that those who are in a content group with access to
+        a restricted unit are able to see that unit in lms.
+        """
+        self.course_outline_page.visit()
+        self.course_outline_page.expand_all_subsections()
+        unit = self.course_outline_page.section_at(0).subsection_at(0).unit_at(0)
+        unit.set_unit_access('Content Groups', [self.content_group_a_id])
+        self.course_outline_page.view_live()
+
+        course_home_page = CourseHomePage(self.browser, self.course_id)
+        course_home_page.visit()
+        course_home_page.resume_course_from_header()
+        self.assertEqual(course_home_page.outline.num_units, 2)
+        course_home_page.visit()
+        course_home_page.preview.set_staff_view_mode("Learner in Test Group A")
```

Comment:
There is a tension with bok choy tests-- normally, it is good to have separate tests to test different scenarios (as you have done here). But bok choy tests are notoriously slow. Therefore we tend to do more things in a single test point to avoid the setup time/cost.

I would put lines 738-741 at the end of the previous test point (and update the name accordingly) since the setup is all the same, and really we are testing what happens when you view as learners in different groups.